extensions/swagger-ui/runtime/src/main/java/io/quarkus/swaggerui/runtime/SwaggerUiRecorder.java

```diff
@@ -13,31 +11,21 @@
 public class SwaggerUiRecorder {
     public Handler<RoutingContext> handler(String swaggerUiFinalDestination, String swaggerUiPath) {
 
-        Handler<RoutingContext> handler = new ThreadLocalHandler(new Supplier<Handler<RoutingContext>>() {
-            @Override
-            public Handler<RoutingContext> get() {
-                return StaticHandler.create().setAllowRootFileSystemAccess(true)
-                        .setWebRoot(swaggerUiFinalDestination)
-                        .setDefaultContentEncoding("UTF-8");
-            }
-        });
-
-        return new Handler<RoutingContext>() {
-            @Override
-            public void handle(RoutingContext event) {
-                if (event.normalisedPath().length() == swaggerUiPath.length()) {
-
-                    event.response().setStatusCode(302);
-                    event.response().headers().set(HttpHeaders.LOCATION, swaggerUiPath + "/");
-                    event.response().end();
-                    return;
-                } else if (event.normalisedPath().length() == swaggerUiPath.length() + 1) {
-                    event.reroute(swaggerUiPath + "/index.html");
-                    return;
-                }
+        Handler<RoutingContext> handler = new ThreadLocalHandler(() -> StaticHandler.create().setAllowRootFileSystemAccess(true)
+                .setWebRoot(swaggerUiFinalDestination)
+                .setDefaultContentEncoding("UTF-8"));
 
-                handler.handle(event);
+        return event -> {
+            if (event.normalisedPath().length() == swaggerUiPath.length()) {
+                event.response().setStatusCode(302);
+                event.response().headers().set(HttpHeaders.LOCATION, swaggerUiPath + "/");
+                event.response().end();
+                return;
+            } else if (event.normalisedPath().length() == swaggerUiPath.length() + 1) {
+                event.reroute(swaggerUiPath + "/index.html");
+                return;
             }
+            handler.handle(event);
```

Comment:
Please don't rewrite code like that as I have absolutely no idea of what you changed.

There is a good reason we don't use lambdas in runtime code: they have a memory cost that we want to avoid.

Reply:
I got it :)